src/lib/core/fiber_channel.c

```diff
@@ -191,7 +206,54 @@ fiber_channel_check_wait(struct fiber_channel *ch, ev_tstamp start_time,
 void
 fiber_channel_close(struct fiber_channel *ch)
 {
-	if (ch->is_closed)
+	/**
+	 * The forceful close mode means that fiber_channel_close()
+	 * discards all stored messages, wake ups all readers and
+	 * writers and forbids to read or write messages. The
+	 * channel is effectively died after closing.
+	 *
+	 * fiber_channel_close() works exactly as
+	 * fiber_channel_destroy() when this mode is in effect.
+	 *
+	 * This mode represents original behavior of the fiber
+	 * channel since its born and hence it is the default.
+	 */
+	if (fiber_channel_close_mode == FIBER_CHANNEL_CLOSE_FORCEFUL)
+		goto destroy;
```

Comment:
Nit: It is strange to see a documentation comment inside a function body.

Also I would leave here only information that helps to understand the algorithm. Mode X do A, B, C. Mode Y do D, E, F.

Reply:
I would also make assumption of the code explicit:

* A channel has either readers or writters (if any).
* If the buffer has any message, the channel can't have a reader.

Reply:
Fixed the aforementioned comment. Also added the new comment for the section below.